hphp/hack/src/errors/errors.ml

```diff
@@ -1790,17 +1641,25 @@ let typeconst_concrete_concrete_override pos parent_pos =
       (parent_pos, "Previously defined here");
     ]
 
-let const_without_typehint sid =
+let const_without_typehint sid type_ =
   let (pos, name) = sid in
-  let msg =
-    Printf.sprintf
-      "Please add a type hint `const SomeType %s`"
-      (Utils.strip_all_ns name)
+  let name = Utils.strip_all_ns name in
+  let msg = Printf.sprintf "Please add a type hint `const SomeType %s`" name in
+  let title, new_text =
+    match type_ with
+    | "string" -> ("Add string type annotation", "string " ^ name)
+    | "int" -> ("Add integer type annotation", "int " ^ name)
+    | "float" -> ("Add float type annotation", "float " ^ name)
+    | _ -> ("Add mixed type annotation", "mixed " ^ name)
   in
-  add (Naming.err_code Naming.AddATypehint) pos msg
+  add_list
+    (Naming.err_code Naming.AddATypehint)
+    ~quickfixes:[ Quickfix.make ~title ~new_text pos ]
+    (pos, msg)
+    [ (Pos_or_decl.none, "Declaration is here") ]
```

Comment:
This list is the extra information we show when there's a type error.

```suggestion
    []
```

For example, if I write a bad program like this:

```hack
function takes_int(int $_): void {}

function call_it(): void {
  takes_int("foo");
}
```

Then the type checker complains:

```
$ hh_client
Typing[4110] Invalid argument [1]
-> Expected int [2]
-> But got string [1]

x.php:8:13
    3 | // @format
    4 | 
[2] 5 | function takes_int(int $_): void {}
    6 | 
    7 | function call_it(): void {
[1] 8 |   takes_int("foo");
    9 | }

1 error found.
```

The "expected int", "but got string" are the extra reasons. The main error message here is "invalid argument".

In your case, there's no extra reasons we need to give.